Add `--orphans` flag for bundle-list

castulo opened the pull request:
This commit adds the `--orphans` flag to `bundle-list` and `3rd-party bundle-list`. This flag will list the bundles that are no longer required by any tracked bundles.

Closes #1530

castulo commented:
@otaviobp this PR is going to fail because it adds tests to the `api` test suite.

otaviobp commented on src/bundle_list.c:
```diff
@@ -419,6 +445,91 @@ static enum swupd_code show_bundle_reqd_by(const char *bundle_name, bool server,
+	tracked_bundles = bundle_list_tracked();
+	if (!tracked_bundles) {
+		// this should never happen, swupd_init makes sure of it
+		UNEXPECTED();
```
The idea when I created the UNEXPECTED macro was for some unexpected behaviors that we try to recover and that wouldn't be able to be noticed unless we have an abort() call to warn us. If it's a situation where you return an error code, you don't need to use that. Just return the error code and the test is going to break.

castulo replied:
got it

otaviobp commented on src/bundle_list.c:
```diff
@@ -419,6 +445,91 @@ static enum swupd_code show_bundle_reqd_by(const char *bundle_name, bool server,
+			print("%s\n", name);
+			FREE(name);
+		} else {
+			UNEXPECTED();
```
This is a good usage of UNEXPECTED()

otaviobp commented on src/bundle_list.c:
```diff
@@ -419,6 +445,91 @@ static enum swupd_code show_bundle_reqd_by(const char *bundle_name, bool server,
+			error("Cannot load included bundles\n\n");
+			ret_code = SWUPD_COULDNT_LOAD_MANIFEST;
+		} else {
+			UNEXPECTED();
```
You don't need this here.

castulo commented:
V2:
- Removed "prefix" from sys_ls()
- Removed get_dir_files_sorted() in favor of using sys_ls()
- Corrected usage of UNEXPECED()
- Fixed a problem with showing a message even if swupd failed to list
- Not using errno to check for errors

otaviobp commented on src/bundle_list.c:
```diff
@@ -476,6 +589,9 @@ enum swupd_code bundle_list_main(int argc, char **argv)
 
 	ret = list_bundles();
+	if (cmdline_option_orphans) {
+		info("\nUse \"swupd bundle-add BUNDLE\" to no longer list BUNDLE and its dependencies as orphaned\n");
```
I still think it doesn't make sense to print this if the number of orphan bundles is 0
